scapy/contrib/gtp.py

```diff
@@ -531,7 +531,10 @@ class IE_GSNAddress(IE_Base):
     name = "GSN Address"
     fields_desc = [ByteEnumField("ietype", 133, IEType),
                    ShortField("length", 4),
```

Comment:
With your change, building  `IE_GSNAddress(ipv6_address="::1")` will use an incorrect length, and will break the dissection.

Reply:
Thank you for catching this!
I have added the post_build function to calculate the length.
The default length is still 4 because there are many places where IE_GSNAddress() is constructed without any argument, and it still creates a valid IE in this case. Hence, the length is calculated only when it is left at its default value.
I appreciate your feedback.